src/topology/algebra/open_subgroup.lean

```diff
@@ -140,6 +140,17 @@ instance : semilattice_inf_top (open_subgroup G) :=
 attribute [norm_cast] coe_inf coe_subset coe_subgroup_le open_add_subgroup.coe_inf
   open_add_subgroup.coe_subset open_add_subgroup.coe_add_subgroup_le
 
+/-- The preimage of an `open_subgroup` along a continuous `monoid` homomorphism is a subgroup. -/
+@[to_additive "The preimage of an `open_add_subgroup` along an `open_add_monoid` homomorphism
+is an `open_add_subgroup`."]
+def comap {N : Type*} [group N] [topological_space N] [topological_group N] (f : G →* N)
```

Comment:
Can you add the lemmas `coe_comap` and `mem_comap`, like the ones that exist for `subgroup`?

Reply:
Done. Do I need to also add `comap_comap`? Or would that be redundant? I'll also add `open_subgroup.map` in another separate PR at some point

Reply:
`comap_comap` would certainly be a nice addition too.

Reply:
By the way, I've left `f` explicit, because Lean has trouble inferring that things are monoid homomorphisms, like for instance if you have `g` and `f` monoid homs and `hg: continuous g` and `hf: continuous f` then Lean doesn't see that  `g.comp f` is a monoid hom when you write `hg.comp hf`. Maybe there's a more elegant fix.